src/net.cpp

```diff
@@ -2525,9 +2526,20 @@ void CConnman::AddNewAddresses(const std::vector<CAddress>& vAddr, const CAddres
     addrman.Add(vAddr, addrFrom, nTimePenalty);
 }
 
-std::vector<CAddress> CConnman::GetAddresses()
+std::vector<CAddress> CConnman::GetAddresses(Optional<Network> requestor_network)
 {
-    return addrman.GetAddr();
+    if (requestor_network) {
+        const std::chrono::microseconds current_time = GetTime<std::chrono::microseconds>();
```

Comment:
This is a good place for an `auto`. The return type of `GetTime<std::chrono::microseconds>()` is obvious, so there's no need to type it out.